### PR TITLE
feat(workspace): add a new resource workspace eip associate

### PR DESCRIPTION
**Pull request by wuzhuanhong**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST=./huaweicloud/services/acceptance/workspace TESTARGS='-run TestAccEipAssociate_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/workspace -v -run TestAccEipAssociate_basic -timeout 360m -parallel 4
=== RUN   TestAccEipAssociate_basic
=== PAUSE TestAccEipAssociate_basic
=== CONT  TestAccEipAssociate_basic

--- PASS: TestAccEipAssociate_basic (1104.69s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/workspace       1104.733s
```

### CODE REVIEW

File: docs/resources/workspace_eip_associate.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+* `eip_id` - (Required, String, ForceNew) Specifies the EIP ID to associate. Changing this will create a new resource.
+
+## Attribute Reference
+
+In addition to all arguments above, the following attributes are exported:
+
+* `id` - The associate EIP in UUID format.
+
+* `enterprise_project_id` - The enterprise project ID to which associate the EIP.
```
Comment (Lance52259) [inline docs/resources/workspace_eip_associate.md]:
```suggestion
* `enterprise_project_id` - The enterprise project ID to which the EIP associated.
```

File: docs/resources/workspace_eip_associate.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+* `public_ip` - The address of the EIP.
+
+## Import
+
+EIP association can be imported using the `id` of the EIP, e.g.
```
Comment (Lance52259) [inline docs/resources/workspace_eip_associate.md]:
```suggestion
EIP association can be imported using the `desktop_id` and associated `eip_id`, separated by a slash, e.g.
```

File: docs/resources/workspace_eip_associate.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+* `desktop_id` - (Required, String, ForceNew) Specifies the desktop ID. Changing this will create a new resource.
+
+* `eip_id` - (Required, String, ForceNew) Specifies the EIP ID to associate. Changing this will create a new resource.
+
+## Attribute Reference
+
+In addition to all arguments above, the following attributes are exported:
+
+* `id` - The associate EIP in UUID format.
```
Comment (Lance52259) [inline docs/resources/workspace_eip_associate.md]:
```suggestion
* `id` - The resource ID, the format is `<desktop_id>/<eip_id>`.
```

File: docs/resources/workspace_eip_associate.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
+---
+subcategory: "Workspace"
+---
+
+# huaweicloud_workspace_eip_associate
+
+Associates an EIP to a specified desktop.
```
Comment (Lance52259) [inline docs/resources/workspace_eip_associate.md]:
```suggestion
Using this resource to associate an EIP to a specified desktop within HuaweiCloud.
```

File: huaweicloud/services/acceptance/workspace/resource_huaweicloud_workspace_eip_associate_test.go [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/common"
+)
+
+func getEipAssociateFunc(conf *config.Config, state *terraform.ResourceState) (interface{}, error) {
+	client, err := conf.WorkspaceV2Client(acceptance.HW_REGION_NAME)
+	if err != nil {
+		return nil, fmt.Errorf("Error creating Workspace v2 client: %s", err)
```
Comment (Lance52259) [inline huaweicloud/services/acceptance/workspace/resource_huaweicloud_workspace_eip_associate_test.go]:
```suggestion
		return nil, fmt.Errorf("error creating Workspace v2 client: %s", err)
```

File: huaweicloud/services/acceptance/workspace/resource_huaweicloud_workspace_eip_associate_test.go [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+		PreCheck:          func() { acceptance.TestAccPreCheck(t) },
+		ProviderFactories: acceptance.TestAccProviderFactories,
+		CheckDestroy:      associateRc.CheckResourceDestroy(),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccEipAssociate_base(name),
+				Check: resource.ComposeTestCheckFunc(
+					rc.CheckResourceExists(),
+				),
+			},
```
Comment (Lance52259) [inline huaweicloud/services/acceptance/workspace/resource_huaweicloud_workspace_eip_associate_test.go]:
```suggestion
func TestAccEipAssociate_basic(t *testing.T) {
	var (
		eips         []desktops.EipResp
		resourceName = "huaweicloud_workspace_eip_associate.test"
		name         = acceptance.RandomAccResourceNameWithDash()
		rc           = acceptance.InitResourceCheck(resourceName, &eips, getEipAssociateFunc)
	)

	resource.ParallelTest(t, resource.TestCase{
		PreCheck:          func() { acceptance.TestAccPreCheck(t) },
		ProviderFactories: acceptance.TestAccProviderFactories,
		CheckDestroy:      rc.CheckResourceDestroy(),
		Steps: []resource.TestStep{
```

File: huaweicloud/services/acceptance/workspace/resource_huaweicloud_workspace_eip_associate_test.go [hunk 7]
```diff
@@ -0,0 +1,174 @@
...
+
+  nic {
+    network_id = huaweicloud_vpc_subnet.test.id
+  }
+
+  name       = "%[2]s-1"
+  user_name  = "user-%[2]s"
+  user_email  = "terraform@example.com"
+  user_group  = "administrators"
+  delete_user = true
```
Comment (Lance52259) [inline huaweicloud/services/acceptance/workspace/resource_huaweicloud_workspace_eip_associate_test.go]:
```suggestion
  name        = "%[2]s-1"
  user_name   = "user-%[2]s"
  user_email  = "terraform@example.com"
  user_group  = "administrators"
  delete_user = true
```

File: docs/resources/workspace_eip_associate.md (hunk 1 above)
Comment (wuzhuanhong) [reply docs/resources/workspace_eip_associate.md]: fixed

File: docs/resources/workspace_eip_associate.md (hunk 2 above)
Comment (wuzhuanhong) [reply docs/resources/workspace_eip_associate.md]: fixed

Comment (wuzhuanhong) [reply docs/resources/workspace_eip_associate.md]: fixed

File: docs/resources/workspace_eip_associate.md (hunk 3 above)
Comment (wuzhuanhong) [reply docs/resources/workspace_eip_associate.md]: fixed

File: docs/resources/workspace_eip_associate.md (hunk 4 above)
Comment (wuzhuanhong) [reply docs/resources/workspace_eip_associate.md]: fixed

File: huaweicloud/services/acceptance/workspace/resource_huaweicloud_workspace_eip_associate_test.go (hunk 5 above)
Comment (wuzhuanhong) [reply huaweicloud/services/acceptance/workspace/resource_huaweicloud_workspace_eip_associate_test.go]: fixed

File: huaweicloud/services/acceptance/workspace/resource_huaweicloud_workspace_eip_associate_test.go (hunk 6 above)
Comment (wuzhuanhong) [reply huaweicloud/services/acceptance/workspace/resource_huaweicloud_workspace_eip_associate_test.go]: fixed

File: huaweicloud/services/acceptance/workspace/resource_huaweicloud_workspace_eip_associate_test.go (hunk 7 above)
Comment (wuzhuanhong) [reply huaweicloud/services/acceptance/workspace/resource_huaweicloud_workspace_eip_associate_test.go]: fixed

**Comment by wuzhuanhong** [comment]
make testacc TEST=./huaweicloud/services/acceptance/workspace TESTARGS='-run TestAccEipAssociate_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/workspace -v -run TestAccEipAssociate_basic -timeout 360m -parallel 4
=== RUN   TestAccEipAssociate_basic
=== PAUSE TestAccEipAssociate_basic
=== CONT  TestAccEipAssociate_basic
--- PASS: TestAccEipAssociate_basic (994.85s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/workspace   994.909s

**Comment by wuzhuanhong** [comment]
make testacc TEST=./huaweicloud/services/acceptance/workspace TESTARGS='-run TestAccEipAssociate_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/workspace -v -run TestAccEipAssociate_basic -timeout 360m -parallel 4
=== RUN   TestAccEipAssociate_basic
=== PAUSE TestAccEipAssociate_basic
=== CONT  TestAccEipAssociate_basic
--- PASS: TestAccEipAssociate_basic (945.14s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/workspace 945.191s

File: huaweicloud/services/workspace/resource_huaweicloud_workspace_eip_associate.go [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+
+	mErr := multierror.Append(nil, d.Set("region", region))
+	if len(eips) > 0 {
+		mErr = multierror.Append(mErr,
+			d.Set("desktop_id", eips[0].AttachedDesktopId),
+			d.Set("eip_id", eips[0].ID),
+			d.Set("enterprise_project_id", eips[0].EnterpriseProjectId),
+			d.Set("public_ip", eips[0].Address),
+		)
+	}
```
Comment (Lance52259) [inline huaweicloud/services/workspace/resource_huaweicloud_workspace_eip_associate.go]:
```suggestion
	if len(eips) < 1 {
		return common.CheckDeletedDiag(d, golangsdk.ErrDefault404{}, "Workspace EIP associate")
	}
	associatedEipInfo := eips[0]
	mErr := multierror.Append(nil,
		d.Set("region", region),
		d.Set("desktop_id", associatedEipInfo.AttachedDesktopId),
		d.Set("eip_id", associatedEipInfo.ID),
		d.Set("enterprise_project_id", associatedEipInfo.EnterpriseProjectId),
		d.Set("public_ip", associatedEipInfo.Address),
	)
```

Comment (Lance52259) [reply huaweicloud/services/workspace/resource_huaweicloud_workspace_eip_associate.go]: If the List method cannot obtain the associated EIP information, please enter the CheckDeleted logic

Comment (wuzhuanhong) [reply huaweicloud/services/workspace/resource_huaweicloud_workspace_eip_associate.go]: make testacc TEST=./huaweicloud/services/acceptance/workspace TESTARGS='-run TestAccEipAssociate_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/workspace -v -run TestAccEipAssociate_basic -timeout 360m -parallel 4
=== RUN   TestAccEipAssociate_basic
=== PAUSE TestAccEipAssociate_basic
=== CONT  TestAccEipAssociate_basic
--- PASS: TestAccEipAssociate_basic (1122.86s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/workspace 1122.904s

**Pull request merged**